app/components/primer/button_component.html.erb

```diff
@@ -0,0 +1,12 @@
+<%= render Primer::BaseButton.new(**@system_arguments) do %>
+  <% if icon.present? && @icon_align == :left %>
+    <%= icon %>
+  <% end %>
+
+  <%= content %>
+  <%= counter %>
```

Comment:
@vdepizzol is it valid to have a `counter` _and_ and icon to the right of a the content of a button? IMO we should ensure this isn't the case.

Reply:
@joelhawksley I'm trying to visualize which cases where a right-aligned icon makes sense — rendering a `triangle-down` to indicate the button opens a menu or an overlay being the only example I can think of. And that feels like a boolean option that could appear even if the button already has a left-positioned icon. Is that something that could be supported with the icon alignment considerations from this PR?

Example:
> ![Screen Shot 2021-04-23 at 15 31 43](https://user-images.githubusercontent.com/293280/115935917-0d70db80-a449-11eb-908c-1081096a00fe.png)

For a counter + right icon scenario, having it with a dropdown makes sense to me. Like in the example below:

> ![Screen Shot 2021-04-23 at 15 35 33](https://user-images.githubusercontent.com/293280/115936265-a0117a80-a449-11eb-9c62-fa5b6152273e.png)

@manuelpuyol are there previously discussed use cases of a right-aligned icon that I'm missing? ✨ 

Thanks! 🙇 🙇

Reply:
I just rolled with a comment made when I asked about it:

> We have icons on the right sometimes

I can't imagine any other icon other than the caret being right-aligned, though. @colebemis, do you know of any other use cases? I'm down to always aligning left and having a `caret` flag to render the caret

Reply:
@manuelpuyol I'd vote to start with the more restrictive case (left alignment, caret flag). We can always make the API more permissive later 😄